### PR TITLE
Add explict provider test to conformance tests

### PR DESCRIPTION
**Pull request by Frassle**
# Description

Adds a conformance test for an explicit provider resource.

### CODE REVIEW

File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/projects/l2-explicit-provider/__main__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import pulumi
+import pulumi_simple as simple
+
+prov = simple.Provider("prov")
+res = simple.Resource("res", value=True,
+opts=pulumi.ResourceOptions(provider=prov))
```
Comment (lunaris) [inline sdk/python/cmd/pulumi-language-python/testdata/setuppy/projects/l2-explicit-provider/__main__.py:6]:
```suggestion
opts = pulumi.ResourceOptions(provider=prov))
```

File: sdk/python/cmd/pulumi-language-python/testdata/toml/projects/l2-explicit-provider/__main__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import pulumi
+import pulumi_simple as simple
+
+prov = simple.Provider("prov")
+res = simple.Resource("res", value=True,
+opts=pulumi.ResourceOptions(provider=prov))
```
Comment (lunaris) [inline sdk/python/cmd/pulumi-language-python/testdata/toml/projects/l2-explicit-provider/__main__.py:6]:
```suggestion
opts = pulumi.ResourceOptions(provider=prov))
```

**Review by lunaris — COMMENTED** [review]
LGTM! Noob question -- we don't currently do conformance testing in languages other than NodeJS/Python I'm guessing?

File: cmd/pulumi-test-language/tests.go [hunk 3]
```diff
@@ -296,6 +296,34 @@ var languageTests = map[string]languageTest{
...
+		providers: []plugin.Provider{&providers.SimpleProvider{}},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, res result.Result,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					requireStackResource(l, res, changes)
+
+					// Check we have the one simple resource in the snapshot, it's provider and the stack.
```
Comment (lunaris) [inline cmd/pulumi-test-language/tests.go:309]:
```suggestion
					// Check we have the one simple resource in the snapshot, its provider and the stack.
```

**Review by lunaris — APPROVED** [review]

**Review by julienp — APPROVED** [review]

**Pull request merged**